test/cpp/microbenchmarks/bm_cq.cc

```diff
@@ -205,8 +210,15 @@ static void BM_Callback_CQ_Pass1Core(benchmark::State& state) {
   gpr_cv_init(&shutdown_cv);
   bool got_shutdown = false;
   ShutdownCallback shutdown_cb(&got_shutdown);
-  grpc_completion_queue* cc =
-      grpc_completion_queue_create_for_callback(&shutdown_cb, nullptr);
```

Comment:
Why on heap we could use grpc_completion_queue_create_for_callback(), but not on stack?

Reply:
So in general, the callback could get triggered on a different thread, in which case the stack might have already deallocated those items by that point. An alternative would have been to put a CV into the stack version but then there would only be 1 CV outstanding at a time which would be contradictory to the purpose of this microbenchmark (testing the overhead of the CQ).

Reply:
I updated the comment to make this more clear.

Reply:
I see, thanks for the explanation. So the actual difference is in how the completion queue and the polling engine interact?  

(grpc_completion_queue_create_for_callback() is using GRPC_CQ_DEFAULT_POLLING, which allows other worker threads to do polling. Should we consider change grpc_completion_queue_create_for_callback() to use GRPC_CQ_NON_POLLING instead? )